File: solutions/067.add-binary/add-binary.js
/**
 * @param {string} a
 * @param {string} b
 * @return {string}
 */
var addBinary = function(a, b) {
    let res = '';
    let carry = 0, i = a.length - 1, j = b.length - 1;
    while(i >= 0 || j >= 0 || carry !== 0){
        let digit1 = i >= 0 ? +a[i--] : 0;
        let digit2 = j >= 0 ? +b[j--] : 0;
        const sum = digit1 + digit2 + carry;
        res = (sum % 2) + res;
        carry = Math.floor(sum / 2);
    }
    return res;
};

File: solutions/067.add-binary/add-binary.java
class Solution {
    public String addBinary(String a, String b) {
        StringBuilder res = new StringBuilder();
        int toAdd = 0;
        int lenA = a.length(), lenB = b.length();
        int len = Math.max(lenA, lenB);
        for(int i = 0; i < len; i++){
            char charA = i < lenA ? a.charAt(lenA - 1- i) : '0';
            char charB = i < lenB ? b.charAt(lenB -1 - i) : '0';
            int sum = charA - '0' + charB - '0' +toAdd;
            res.insert(0, sum % 2);
            toAdd = sum / 2;
        }
       
        if(toAdd != 0){
            res.insert(0, toAdd);
        }
        return res.toString();
    }
}